x/concentrated-liquidity/tick.go

```diff
@@ -94,7 +94,23 @@ func (k Keeper) getTickInfo(ctx sdk.Context, poolId uint64, tickIndex int64) (ti
 			return tickStruct, err
 		}
 
-		return model.TickInfo{LiquidityGross: sdk.ZeroDec(), LiquidityNet: sdk.ZeroDec(), FeeGrowthOutside: initialFeeGrowthOutside}, nil
+		// Sync global uptime accumulators to ensure the uptime tracker init values are up to date.
+		if err := k.updateUptimeAccumulatorsToNow(ctx, poolId); err != nil {
```

Comment:
I think we should think about when we want to run `updateUptimeAccumulatorsToNow`, doing this every time we call `getTickInfo` definitely feels like an overhead. Suggested way is creating a secondary method `getTickInfoAndUpdateAccumulators` and use this method for the necessary places. If this is not possible, we should at lest update this method name since its not a `get` method any more

Reply:
This code only executes when the tick is being initialized (note that it's under `if !found`), not on every call

Reply:
Unresolving as I have a follow on question:

As Matt said, shouldn't this not be named "get" since it has the potential to actually initialize the tick accum? Maybe we can check if !found prior, and if it isn't then we call and `initTickInfo` which rolls into a `getTickInfo`. Otherwise we just call `getTickInfo`.